streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -151,7 +187,16 @@
 
     @Override
     public KTable<K, Long> count(final String storeName) {
-        return count(keyValueStore(keySerde, Serdes.Long(), storeName));
+        final String internalStoreName = storeName != null ? storeName : topology.newStoreName(AGGREGATE_NAME);
+        Topic.validate(internalStoreName);
```

Comment:
We should validate everywhere -- IMHO, everywhere is better than in a central place to avoid deep stacktraces.